### ISSUE TITLE
Cypress E2E tests cite uncaught promise in Inertia Vue 3 adapter code

### ISSUE THREAD

**Issue by plweil** [inertiajs/inertia#1622]
### Version:

- `@inertiajs/vue3` version: 1.09

### Describe the problem:

When attempting to run E2E tests with Cypress, Cypress is unable to actually run any specs(or so it appears), instead returns an error

```
Uncaught (in promise) TypeError: h2 is null
```

and points to this line in the Inertia Vue 3 adapter:

1. As displayed in Chrome devtools/Source:

```javascript
async function j$1({ id: e2 = "app", resolve: o2, setup: n2, title: p2, progress: t2 = {}, page: a2, render: l2 }) {
    let f2 = typeof window > "u", h2 = f2 ? null : document.getElementById(e2), y2 = a2 || JSON.parse(h2.dataset.page), r2 = (c2) => Promise.resolve(o2(c2)).then((b2) => b2.default || b2), s2 = [], i2 = await r2(y2.component).then((c2) => n2({ el: h2, App: I$1, props: { initialPage: y2, initialComponent: c2, resolveComponent: r2, titleCallback: p2, onHeadUpdate: f2 ? (b2) => s2 = b2 : null }, plugin: $$1 }));
    if (!f2 && t2 && Z$1(t2), f2) {
      let c2 = await l2(createSSRApp({ render: () => h$1("div", { id: e2, "data-page": JSON.stringify(y2), innerHTML: i2 ? l2(i2) : "" }) }));
      return { head: s2, body: c2 };
    }
  }
``` 

2. what appears to be the same function in index.esm.js:

```javascript
async function j({id:e="app",resolve:o,setup:n,title:p,progress:t={},page:a,render:l}){let f=typeof window>"u",h=f?null:document.getElementById(e),y=a||JSON.parse(h.dataset.page),r=c=>Promise.resolve(o(c)).then(b=>b.default||b),s=[],i=await r(y.component).then(c=>n({el:h,App:I,props:{initialPage:y,initialComponent:c,resolveComponent:r,titleCallback:p,onHeadUpdate:f?b=>s=b:null},plugin:$}));if(!f&&t&&Q(t),f){let c=await l(Y({render:()=>z("div",{id:e,"data-page":JSON.stringify(y),innerHTML:i?l(i):""})}));return{head:s,body:c}}

```

### Steps to reproduce:

Run a Cypress E2E test on an application using Laravel, Inertia.js, Vite, and Vue 3 with the Vue 3 adapter.

Cypress v12.17.2
Laravel 9
Vue 3.3.1
Vite 4.3.5
Mac OS 12.7

**Comment by reinink** [comment]
Hey! I'd love to help you out here, but this is too general of an issue for me to help with, especially without a minimal reproduction. If you're able to provide a super minimal reproduction of the issue as a git repo I might be able to help out, but unfortunately without that I can't.

**Comment by plweil** [comment]
Thanks @reinink. This afternoon I was finally able to narrow the issue down to the cypress-vite plugin. It might be that Inertia just got caught in the crossfire, so to speak. I don't know for sure why that line was cited by Cypress, but when I removed the plugin (which I learned that I didn't even need for E2E testing), the problem disappeared. Anyway, this is my first project with inertia and I think it's fantastic. Thank you!
